src/rpc/rawtransaction.cpp

```diff
@@ -922,67 +921,77 @@ static RPCHelpMan testmempoolaccept()
                                              CFeeRate(AmountFromValue(request.params[1]));
 
     CTxMemPool& mempool = EnsureMemPool(request.context);
-    int64_t virtual_size = GetVirtualTransactionSize(*tx);
-    CAmount max_raw_tx_fee = max_raw_tx_fee_rate.GetFee(virtual_size);
-
-    UniValue result(UniValue::VARR);
-    UniValue result_0(UniValue::VOBJ);
-    result_0.pushKV("txid", tx->GetHash().GetHex());
-    result_0.pushKV("wtxid", tx->GetWitnessHash().GetHex());
+    std::vector<CTransactionRef> txns;
 
-    TxValidationState state;
-    bool test_accept_res;
-    CAmount fee{0};
-    {
-        LOCK(cs_main);
-        test_accept_res = AcceptToMemoryPool(mempool, state, std::move(tx),
-            nullptr /* plTxnReplaced */, false /* bypass_limits */, /* test_accept */ true, &fee);
+    for (unsigned int i = 0; i < raw_transactions.size(); ++i) {
+        CMutableTransaction mtx;
+        if (!DecodeHexTx(mtx, raw_transactions[i].get_str())) {
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+        }
+        txns.emplace_back(MakeTransactionRef(std::move(mtx)));
     }
 
-    // Check that fee does not exceed maximum fee
-    if (test_accept_res && max_raw_tx_fee && fee > max_raw_tx_fee) {
-        result_0.pushKV("allowed", false);
-        result_0.pushKV("reject-reason", "max-fee-exceeded");
-        result.push_back(std::move(result_0));
-        return result;
+    std::vector<MempoolAcceptResult> validation_results;
+    if (txns.size() == 1) {
+        validation_results.emplace_back(WITH_LOCK(cs_main,
+            return AcceptToMemoryPool(mempool, txns[0], false /* bypass_limits */, true /* test_accept */)));
+    } else {
+        validation_results = WITH_LOCK(cs_main, return ProcessNewPackage(mempool, txns, true));
     }
-    result_0.pushKV("allowed", test_accept_res);
 
-    // Only return the fee and vsize if the transaction would pass ATMP.
-    // These can be used to calculate the feerate.
-    if (test_accept_res) {
-        result_0.pushKV("vsize", virtual_size);
-        UniValue fees(UniValue::VOBJ);
-        fees.pushKV("base", ValueFromAmount(fee));
-        result_0.pushKV("fees", fees);
-    } else {
-        if (state.IsInvalid()) {
-            if (state.GetResult() == TxValidationResult::TX_MISSING_INPUTS) {
-                result_0.pushKV("reject-reason", "missing-inputs");
+    auto tx_accepted = [](MempoolAcceptResult& res) { return res.m_accepted; };
+    const bool all_valid = std::all_of(validation_results.begin(), validation_results.end(), tx_accepted);
+    // ProcessNewPackage should return a MempoolAcceptResult per transaction
+    // or exactly 1 for the first error that occurs.
+    CHECK_NONFATAL(all_valid || validation_results.size() == 1);
+
+    // TODO: Report absurd fees for packages.
+
+    UniValue result(UniValue::VARR);
+    if (all_valid) {
+        for (unsigned int i = 0; i < validation_results.size(); ++i) {
```

Comment:
Just confirming, does the requirement "either return all successes or just the first failure" apply to the RPC as a whole, or only to the internal call `ProcessNewPackage`?

I ask because it looks like you could add rejected transactions inside this loop. For example, if I update `test_chain` in `rpc_packages.py` so that the 2nd transaction in the chain has an unreasonable fee, the RPC response `testres_multiple` is as follows:

```
[
{'txid': ..., 'wtxid': ..., 'allowed': True, 'vsize': 191, 'fees': {'base': Decimal('0.00010000')}},
{'txid': ..., 'wtxid': ..., 'allowed': False, 'reject-reason': 'max-fee-exceeded'},
{'txid': ..., 'wtxid': ..., 'allowed': True, 'vsize': 191, 'fees': {'base': Decimal('0.00010000')}}
]
```

Reply:
it applies to the RPC as well, only 1 result on failure